utils/convertToWeiSafely.ts

```diff
@@ -0,0 +1,21 @@
+import { utils } from "ethers";
+const { parseUnits: toWei } = utils;
+
+const DEFAULT_PRECISION = 18;
+
+// `toWeiSafe()` should always be used to convert floats into wei values
+// before passing the result as a transaction arg, as Solidity cannot deal with non-Integers.
+// If the argument to pass into `toWei()` has too much precision (specifically more than `precisionToUse`),
+// then `toWei()` might return a string number with decimals, which Solidity cannot handle.
+export function toWeiSafe(
+  numberToConvertToWei: string,
+  desiredPrecision?: number
+) {
+  const precisionToUse = desiredPrecision
+    ? desiredPrecision
+    : DEFAULT_PRECISION;
+  return toWei(
```

Comment:
here, where you feed in the precision to use, is this notation only available for `ethers`? can you do this with web3.js as well?

Reply:
Actually this doesn't work for `web3` because `web3` only allows you to specify `ethers` or `gwei` etc. Look at this PR i made in `protocol` where we had to import part of the ethers package to get this functionality for our `web3` repo: https://github.com/UMAprotocol/protocol/pull/1899